src/Symfony/Bridge/Doctrine/Form/Type/EntityType.php

```diff
@@ -18,70 +18,17 @@
 class EntityType extends DoctrineType
 {
     /**
-     * @var ORMQueryBuilderLoader[]
-     */
-    private $loaderCache = array();
-
-    /**
      * Return the default loader object.
      *
-     * @param ObjectManager $manager
-     * @param mixed         $queryBuilder
-     * @param string        $class
+     * @param ObjectManager         $manager
+     * @param QueryBuilder|\Closure $queryBuilder
+     * @param string                $class
      *
      * @return ORMQueryBuilderLoader
      */
     public function getLoader(ObjectManager $manager, $queryBuilder, $class)
```

Comment:
Nvm, since there is not common interface for doctrine query builders for both ORM and ODM, we cannot add a typehint I guess.